webapp/src/components/ResourceMetadata.tsx

```diff
@@ -23,7 +44,31 @@ export default class ResourceMetadata extends React.Component<IPropTypes, never>
       title,
     } = this.props.resource;
 
+    const showConfirmArchiveResourceModal = () => this.setState({ modal: MODAL_TYPES.CONFIRM_ARCHIVE_RESOURCE });
+
+    const deleteSelectedResourceLocally = () => {
+      this.props.deleteResourcesLocally(Array(this.props.resource));
+      this.setState({modal: MODAL_TYPES.NONE});
+    };
+
+    const closeModal = () => this.setState({ modal: MODAL_TYPES.NONE });
+
+    
+
+    let modal;
+      switch (this.state.modal) {
+        case MODAL_TYPES.CONFIRM_ARCHIVE_RESOURCE:
```

Comment:
Can we please put this in `ResourcePage.tsx`? I'd like to avoid having too many places displaying modals. We also want to be displaying the modals as high up in the DOM as possible to avoid CSS changes breaking them. (Modals rely on being able to set `position: absolute` and having that all be relative to the full width and height of the page. If a div containing the modal were to set `position: relative`, it would break the appearance of the modal. The higher up in the DOM the modal is, the lower the likelihood of someone setting `position: relative` on a containing div.)